shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -1173,28 +1272,27 @@ class JSTestBackend extends JSBackend(allowUnresolvedSymbols = false) {
   }
 
   private def generateNewDef(pgrm: Pgrm)(implicit scope: Scope, allowEscape: Bool): JSTestBackend.TestCode = {
-    val mlsModule = topLevelScope.declareValue("typing_unit", Some(false), false)
     val (typeDefs, otherStmts) = pgrm.tops.partitionMap {
       case ot: Terms => R(ot)
       case fd: NuFunDef => R(fd)
       case nd: NuTypeDef => L(nd)
       case _ => die
-   }
+    }
 
-    val (traitSymbols, classSymbols, mixinSymbols, moduleSymbols) = declareNewTypeDefs(typeDefs)
-    val defStmts = 
-      traitSymbols.map { translateTraitDeclaration(_)(topLevelScope) } ++
-      mixinSymbols.map { translateMixinDeclaration(_)(topLevelScope) } ++
-      moduleSymbols.map { translateModuleDeclaration(_)(topLevelScope) } ++
-      classSymbols.map { translateNewClassDeclaration(_)(topLevelScope) }
+    // don't pass `otherStmts` to the top module, because we need to execute them one by one later
```

Comment:
"Top module" doesn't mean anything. Do you mean "top-level"? If so please rename and update the doc.